parser/src/token.rs

```diff
@@ -412,6 +428,22 @@ impl<'input> Tokenizer<'input> {
                 let (end, float) = self.take_while(start, is_digit);
                 (start, end, Token::FloatLiteral(float.parse().unwrap()))
             }
+            Some((end, 'x')) => {
+                self.bump(); // Skip 'x'
+                let (end, hex) = self.take_while(end.shift('x'), is_hex);
+                match int {
+                    "0" | "-0" => {
+                        let is_positive = int == "0";
+                        match i64_from_hex(hex, is_positive) {
+                            Ok(val) => (start, end, Token::IntLiteral(val)),
+                            Err(err) => return self.error(start, err),
+                        }
+                    }
+                    _ => {
+                        return  self.error(start, HexLiteralWrongPrefix);
```

Comment:
super nit: Extra whitespace after return (just run it through rustfmt if you have it).